internal/services/vorpal.go

```diff
@@ -93,16 +93,21 @@ func executeScan(vorpalWrapper grpcs.VorpalWrapper, filePath string) (*grpcs.Sca
 	return vorpalWrapper.Scan(fileName, sourceCode)
 }
 
-func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper) error {
+func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper, vorpalWrappers VorpalWrappersParam) error {
 	vorpalInstalled, _ := osinstaller.FileExists(vorpalconfig.Params.ExecutableFilePath())
 
-	if vorpalParams.VorpalUpdateVersion || !vorpalInstalled {
-		if err := vorpalWrapper.HealthCheck(); err == nil {
+	if !vorpalInstalled || vorpalParams.VorpalUpdateVersion {
+		if err := checkLicense(vorpalParams.IsDefaultAgent, vorpalWrappers); err != nil {
 			_ = vorpalWrapper.ShutDown()
+			return err
 		}
-		if err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params); err != nil {
+		newInstallation, err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params)
+		if err != nil {
 			return err
 		}
+		if newInstallation {
+			_ = vorpalWrapper.ShutDown()
```

Comment:
we are doing a shutdown in other flow before we are running a new one no? just need to make sure there are no bugs in this flow.
let's have tests please..we have many scenarios

Reply:
This function is the only place we are shutting down. This pr did not change this logic

Reply:
we didn't have shutdown aywehre else?

Reply:
same here for the shutdown and healthcheck

Reply:
No, this is the only function that we use shutdowns. 
For healthcheck there are more places